Popular_Models_Guide/StableDiffusion/README.md

```diff
@@ -0,0 +1,253 @@
+[stable-diffusion-v1-5](https://huggingface.co/runwayml/stable-diffusion-v1-5),
+[stable-diffusion-xl](https://huggingface.co/docs/diffusers/en/using-diffusers/sdxl). For
+more information on the TensorRT implementation please see the [TensorRT demo](https://github.com/NVIDIA/TensorRT/tree/release/9.2/demo/Diffusion).
+
+> [!Note]
+> This example is given as sample code and should be reviewed before use in production settings.
+
+| [Requirements](#requirements) | [Building Server Image](#building-the-triton-inference-server-image) | [Stable Diffusion v1.5](#building-and-running-stable-diffusion-v-15) | [Stable Diffusion XL](#building-and-running-stable-diffusion-xl) | [Sending an Inference Request](#sending-an-inference-request) | [Model Configuration](docs/model_configuration.md) | [Client Application](docs/client_application.md) |
+
+## Requirements
+
+The following instructions require a linux system with Docker
+installed. For CUDA support, make sure your CUDA driver meets the
+requirements in "NVIDIA Driver" section of [Deep Learning Framework
+support matrix](https://docs.nvidia.com/deeplearning/frameworks/support-matrix/index.html).
+
+## Building the Triton Inference Server Image
+
+The example is designed based on the
+`nvcr.io/nvidia/tritonserver:24.01-py3` docker image and [TensorRT OSS v9.2.0](https://github.com/NVIDIA/TensorRT/releases/tag/v9.2.0).
+
+A set of convenience scripts are provided to create a docker image
+based on the `nvcr.io/nvidia/tritonserver:24.01-py3` image with the
+dependencies for the TensorRT Stable Diffusion demo installed.
+
+### Trition Inference Server + TensorRT OSS
+
+#### Clone Repository
+```bash
+git clone https://github.com/triton-inference-server/tutorials.git -b nnshah1-stable-diffusion --single-branch
+cd tutorials/Popular_Models_Guide/StableDiffusion
+```
+
+#### Build `tritonserver:r24.01-diffusion` Image
+```bash
+./build.sh
+```
+
+#### Included Models
+
+The `default` build includes model configuration files for
+[`stable_diffusion_1_5`](diffusion-models/stable_diffustion_1_5) and
+[`stable_diffusion_xl`](diffusion-models/stable_diffustion_xl) but the
+actual model artifacts and engine files are not included. They are
+built in a separate step.
+
+
+## Building and Running Stable Diffusion v 1.5
+
+### Start `tritonserver:r24.01-diffusion` Container
+
+The following command starts a container and volume mounts the current
+directory as `workspace`.
+
+```bash
+./run.sh
+```
+
+### Build Stable Diffusion v 1.5 Engine
+
+```bash
+./scripts/build_models.sh --model stable_diffusion_1_5
+```
+
+#### Expected Output
+```
+ diffusion-models
+|-- stable_diffusion_1_5
+|   |-- 1
+|   |   |-- 1.5-engine-batch-size-1
+|   |   |-- 1.5-onnx
+|   |   |-- 1.5-pytorch_model
+|   `-- config.pbtxt
+
+```
+
+### Start a Server Instance
+
+> [!Note]
+> We use `EXPLICIT` model control mode for demonstrtion purposes to control which stable diffusion version is loaded.
```

Comment:
What're we demonstrating with explicit mode in this tutorial? I don't see load/unload calls being used, but might be missing it.

Reply:
Is it just to avoid loading both models due to resource constraints if they had both models setup in the same model repository?

Reply:
because of build times and gpu memory wanted to avoid loading both at once - but wanted to keep both in the repo as samples. As an aside - having support for explicit load _only_ I think is useful :) - so no dynamic load but you can specify the set you want when launching ....

Reply:
Agreed I think `--load-model` could be extended to default mode. Maybe a good new hire task? 

https://github.com/triton-inference-server/core/blob/d4fab30f3a946d242f3fad0f0690c511b916c282/src/model_repository_manager/model_repository_manager.cc#L422-L436